### DISCUSSION TITLE
Confusion/inconsistency re the quality settings in the new Audio Settings preferences dialog and Audio Settings toolbar

### DISCUSSION THREAD

**Discussion by petersampsonaudacity** [audacity/audacity#4591] [Ideas & Feature Requests]
I am finding the quality settings in the new upcoming 3.3.0 Audio Settings dialog confusing.  Well, more accurately, I understand how they work but I am finding it hard to explain the interplay between the three settings to other people (in particular my Mystery-Shopper).  And note the inconsistency between two settings for Sample Rate and just a single one for Sample Format.  

Note too this is no longer something "hidden away" in preferences - a similar dialog exists with greater user accessibility in the **Audio Settings** toolbar dialog.
![image](https://user-images.githubusercontent.com/50205984/233782089-88c224a2-d73c-4408-bf53-21c848fdc0ae.png)

1) for sample rate we have two settings
a) **Project Sample Rate**
The PSR when changed is applied immediately to the current project and is stored with the the project to be re-used for that project when it is reopened

b) **Default Sample Rate**
The DSR has no impact on the currently open project.  It operates as the start-out setting when the user opens a new project or when Audacity is next launched.

2) In contrast for Sample Format we have just the one setting **Default Sample Format**
Like DSR this is the Sample Format that will be used next time Audacity is launched or a new project is opened
But unlike DSR - changing the DSF also has the immediate effect of changing the Sample Format in the currently open project.
But note that the DST is _**NOT**_ stored as part of the project for re-use on reopening the project

So there is an inconsistency in the way the Sample rate and Sample Format work (and this, for me, is where the confusion can lie)

### Proposal
I would suggest that we either have

i) just the two options **Sample Rate** and **Sample Format**
These would both act immediately on the current project and both would be used as the defaults on Audacity relaunch and new project opening.

ii) four options:  **Project Sample Rate**, **Default Sample Rate**, **Project Sample Format**, **Default Sample Format Rate**.
Where the "Project" settings apply to current project and "Default" ones are used on relaunch/open new project.

In both cases I feel the current Sample Rate for the project should also be stored as part of the project for re-use the next time the project is re-opened.

I tend to favor option i) as it looks to be the simpler of the two _(less clutter in the two dialogs that control this)_

---------------------------------------------------------------------------------------------------

To some extent this is a problem that we have created for 3.3.0 as previously the two "Default" settings were tucked away in preferences - and the **Project Rate** was prominently on Display in the old **Selection Toolbar**.

But now all three are co-sited and given more prominence by also being in the **Audio Settings** dialog accessed from the **Audio Setup** toolbar.

I realize that it is probably far to late now for this to be changed for the impending 3.3.0 - but I really think we should seriously consider changing this.

Flagging @Tantacrul @DilsonsPickles and @LWinterberg  as a design/UX issue.

**Comment by petersampsonaudacity** [comment]
I have documented these settings the best I can in the Reference Manual:

![image](https://user-images.githubusercontent.com/50205984/233784306-9127549a-eb15-4743-9a58-174ff30431a1.png)

This is the same in
a) https://alphamanual.audacityteam.org/man/Audio_Setup_Toolbar
b) https://alphamanual.audacityteam.org/man/Audio_Settings_Preferences

**Comment by SteveDaulton** [comment]
Where does the "?" help button go to? (I'm using the cc24ce  AppImage, so the link doesn't open)

**Reply by petersampsonaudacity** [reply]
> Where does the "?" help button go to? (I'm using the cc24ce AppImage, so the link doesn't open)

It doesn't "go" anywhere - it show hovertext when you hover the cursor over it

**Reply by SteveDaulton** [reply]
Then why is it a clickable button?

**Comment by SteveDaulton** [comment]
As an Audacity user, I think it absolutely makes sense to allow the current project to have a different project rate from the default, but I wouldn't expect to find the setting for current project's sample rate to be in Preferences.

Looking at how Audacity handles "defaults" and "current settings" elsewhere:
* Default waveform view is in Preferences. Current waveform view is in the track's options.
The "current setting" may override, but does not change the default setting.
* Audio Host, Preview length, MIDI device, and many other settings are global Preferences.
Changing the global setting affects all projects including the current project.

It's confusing to have a current setting that is not a global setting in Preferences.

**Reply by petersampsonaudacity** [reply]
> ... but I wouldn't expect to find the setting for current project's sample rate to be in Preferences.

More importantly, its more dirctly accessible (and "discoverable") from the **Audio Setup** toolbar:
![image](https://user-images.githubusercontent.com/50205984/233790593-80d86277-d794-4bc5-8c46-8859a5054873.png)

![image](https://user-images.githubusercontent.com/50205984/233790628-dfb92447-4a73-4e5c-b5f4-7211aa78619d.png)

**Reply by SteveDaulton** [reply]
but I still can't see what sample rate my exported file will have without clicking on "Audio Setup" and then clicking on "Audio Settings". I doubt that many new users will "discover" how to set the sample rate of exported files (this was already non-obvious, but now it's non-obvious and non-visible).

**Reply by petersampsonaudacity** [reply]
> but I still can't see what sample rate my exported file will have without clicking on "Audio Setup" and then clicking on "Audio Settings".

I believe that there are likely to be changes afoot with Exports so that you will be able to see and change such settings on export (but _**not**_ for 3.3.0).

**Comment by LWinterberg** [comment]
The current solution is a stop-gap that has many shortcomings. Ideally, the only sample rate to set in this window would be a per-project recording sample rate, with playback rate and default recording rate following the system defaults. And exporting having a sample rate field next to the bit rate.
